File: Core/Entities/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class Employee : BaseEntity
    {
        
        [StringLength(30)]
        public string FirstName { get; set; }
        [StringLength(30)]
        public string LastName { get; set; }
        [StringLength(50)]
        public string Address1 { get; set; }
        [StringLength(50)]
        public string Address2 { get; set; }
        [StringLength(60)]
        public string City { get; set; }
        [StringLength(2)]
        public string State { get; set; }
        [StringLength(15)]
        public string ZipCode { get; set; }
        [StringLength(100)]
        public string Role { get; set; }
        [StringLength(200)]
        public string Department { get; set; }
        [StringLength(1000)]
        public string SkillSets { get; set; }
        public DateTime DateOfBirth { get; set; }

        public DateTime DateOfJoining { get; set; }
        public bool IsActive { get; set; }

    }
}

File: client/src/app/employee.ts
export class Employee {
  id: number;
  firstName: string;
  lastName: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  zipCode: string;
  role: string;
  department: string;
  skillSets: string;
  dateOfBirth: string;
  dateOfJoining: string;
  isActive: boolean;

  constructor(
      id: number,
      firstName: string,
      lastName: string,
      address1: string,
      address2: string,
      city: string,
      state: string,
      zipCode: string,
      role: string,
      department: string,
      skillSets: string,
      dateOfBirth: string,
      dateOfJoining: string,
      isActive: boolean = true)
    {
      this.id = id;
      this.firstName = firstName;
      this.lastName = lastName;
      this.address1 = address1;
      this.address2 = address2;
      this.city = city;
      this.state = state;
      this.zipCode = zipCode;
      this.role = role;
      this.department = department;
      this.skillSets = skillSets;
      this.dateOfBirth = dateOfBirth;
      this.dateOfJoining = dateOfJoining;
      this.isActive = isActive;
    }
}
